cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1874,6 +1879,32 @@ String zkConnectionString(int connectToReplicas, Function<Integer, String> zkNod
             return withVoid(fut);
         }
 
+        /**
+         * Get the cluster Id of the Kafka cluster
+         * 
+         * @return
+         */
+        Future<ReconciliationState> kafkaGetClusterId() {
+            String clusterId = null;
+            log.debug("{}: Attempt to get clusterId", reconciliation);
+            List<String> podNames = IntStream.range(0, kafkaCluster.getReplicas()).boxed().collect(Collectors.toList()).stream().map(podId -> KafkaCluster.kafkaPodName(this.name, podId)).collect(Collectors.toList());
+            try {
+                String bootstrapHostnames = podNames.stream().map(podName -> KafkaCluster.podDnsName(this.namespace, this.name, podName) + ":" + KafkaCluster.REPLICATION_PORT).collect(Collectors.joining(","));
+                log.debug("{}: Creating AdminClient for clusterId using {}", reconciliation, bootstrapHostnames);
+                Admin kafkaAdmin = adminClientProvider.createAdminClient(bootstrapHostnames, this.clusterCa.caCertSecret(), this.oldCoSecret, "cluster-operator");
+                clusterId = kafkaAdmin.describeCluster().clusterId().get();
+            } catch (KafkaException e) {
+                log.debug("{}: Kafka exception getting clusterId {}", reconciliation, e.getMessage());
+            } catch (Exception e) {
```

Comment:
What kind of exception are we expecting here? I think it would be good to use something more specific exception type if possible.

Reply:
As said in the previous comment, the KafkaFuture needs to be handled better anyway; then those exceptions will likely not be a concern anymore.

Reply:
Because of where this is running within `createOrUpdate` I'm guessing it will could fail if the broker aren't ready. Currently that's not really a problem, because the cluster id doesn't change, so most of the time no one would notice. And when creating a cluster the 2nd reconciliation will come around soon enough so the status would only be lacking the id for a short while. If we're going to change the exception handling we'd need to think through the error scenarios in a little more detail. Ignoring the exception in some way is probably the right thing to do, because we don't really want to prevent the rest of the reconciliation from happening. But logging at debug is probably a little too drastic since it's kinda useful for a user to know that we couldn't even describe the cluster. We should go as far as to say the status should be `NotReady` though? @scholzj wdyt?

Reply:
I think there are three separate things. 

First ... the `Exception` seems very generic to me. Catching some more specific types would be better. But I have no clue whether that is possible and what does the Admin API throw. If `Exception` is the best we can do then ok.

Then there are the two issues: How do we log the errors and whether we fail the reconciliation. I agree that the clusterId is not critical enough to fail the reconciliation. But maybe then we should bump the log messages to WARN to make sure they get noticed? Otherwise we might one day find out it never works but nobody notices it. That is at least unless it fails for some reason too often and we do not want to confuse users with the WARN messages.

As for the placement, I didn't noticed it in my review. But it makes sense to have it only after the Kafka pod readiness check.